pyproject.toml

```diff
@@ -9,24 +9,29 @@ show_missing = true
 minversion = "6.0"
 log_cli_level = "INFO"
 
-# Formatting tools configuration
-[tool.black]
+[tool.ruff]
 line-length = 99
-target-version = ["py38"]
 
-[tool.isort]
-line_length = 99
-profile = "black"
+[tool.ruff.lint]
+select = ["E", "W", "F", "C", "N", "D", "I001"]
+extend-ignore = [
+    "D203",
+    "D204",
+    "D213",
+    "D215",
+    "D400",
+    "D404",
+    "D406",
+    "D407",
+    "D408",
+    "D409",
+    "D413",
+]
+ignore = ["E501", "D107"]
+per-file-ignores = {"tests/*" = ["D100","D101","D102","D103","D104"]}
```

Comment:
Seems like we're ignoring much than we used to. Is there any reason for that?

Reply:
we were using the google convention.
The Google convention includes all D errors apart from those on the list. 
I just found out the there is another way of enforcing the convention
```
[tool.ruff.lint.pydocstyle]
convention = "google"
```

Reply:
OK.
What about `E501`? We're setting the desired line length and then we ignore the violations?

Reply:
True, I think that we can disable it only from tests. Like this we can remove all the #noqa
What do you think ?

Reply:
I re enabled the line length again